src/main/java/frc/robot/commands/AlignToPose.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2025 Newport Robotics Group. All Rights Reserved.
+ *
+ * Open Source Software; you can modify and/or share it under the terms of
+ * the license file in the root directory of this project.
+ */
+ 
+package frc.robot.commands;
+
+import com.nrg948.preferences.RobotPreferences.DoubleValue;
+import com.nrg948.preferences.RobotPreferencesLayout;
+import com.nrg948.preferences.RobotPreferencesValue;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.DoubleLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants;
+import frc.robot.subsystems.Subsystems;
+import frc.robot.subsystems.Swerve;
+
+/**
+ * A {@link Command} that autonomous drives and aligns the robot to the specified branch of the
+ * nearest reef side.
+ */
+@RobotPreferencesLayout(groupName = "AlignToPose", row = 0, column = 4, width = 2, height = 3)
+public class AlignToPose extends Command {
+  private static final DataLog LOG = DataLogManager.getLog();
+  private static final double MAX_TRANSLATIONAL_POWER = 0.30;
+  private static final double MAX_ROTATIONAL_POWER = 0.5;
+
+  @RobotPreferencesValue public static DoubleValue Px = new DoubleValue("AlignToReef", "X KP", 1);
+  @RobotPreferencesValue public static DoubleValue Py = new DoubleValue("AlignToReef", "Y KP", 1);
+
+  @RobotPreferencesValue
+  public static DoubleValue Pr = new DoubleValue("AlignToReef", "R KP", 0.02);
+
+  protected final Swerve drivetrain;
+
+  protected Pose2d targetPose;
+
+  private final PIDController xController = new PIDController(Px.getValue(), 0, 0);
+  private final PIDController yController = new PIDController(Py.getValue(), 0, 0);
+  private final PIDController rController = new PIDController(Pr.getValue(), 0, 0);
+
+  private final DoubleLogEntry xErrorLog = new DoubleLogEntry(LOG, "Vision/Reef X Error");
+  private final DoubleLogEntry yErrorLog = new DoubleLogEntry(LOG, "Vision/Reef Y Error");
+  private final DoubleLogEntry rErrorLog = new DoubleLogEntry(LOG, "Vision/Reef Yaw Error");
+
+  private final DoubleLogEntry xTargetLog = new DoubleLogEntry(LOG, "Vision/Reef Target X");
+  private final DoubleLogEntry yTargetLog = new DoubleLogEntry(LOG, "Vision/Reef Target Y");
+  private final DoubleLogEntry rTargetLog = new DoubleLogEntry(LOG, "Vision/Reef Target Yaw");
```

Comment:
Data logging names should match command name and omit "Reef"

```suggestion
  private final DoubleLogEntry xErrorLog = new DoubleLogEntry(LOG, "AlignToPose/X Error");
  private final DoubleLogEntry yErrorLog = new DoubleLogEntry(LOG, "AlignToPose/Y Error");
  private final DoubleLogEntry rErrorLog = new DoubleLogEntry(LOG, "AlignToPose/Yaw Error");

  private final DoubleLogEntry xTargetLog = new DoubleLogEntry(LOG, "AlignToPose/Target X");
  private final DoubleLogEntry yTargetLog = new DoubleLogEntry(LOG, "AlignToPose/Target Y");
  private final DoubleLogEntry rTargetLog = new DoubleLogEntry(LOG, "AlignToPose/Target Yaw");
```